bech32m.py

```diff
@@ -0,0 +1,341 @@
+    return decoded_str
+
+
+def hex_to_base64(hex_str):
+    """
+    convert hex encoded string to base64 encoded string
+    :param hex_str: hexadecimal string to be converted
+    :return: base64 encoded string
+    """
+
+    decoded_str = base64.b64encode(bytes.fromhex(hex_str)).decode()
+    return decoded_str
+
+
+def hex_to_raw(hex_str):
+    """
+    convert hex string to raw bytes
+    :param hex_str:
+    :return: raw bytes
+    """
+
+    return binascii.unhexlify(hex_str)
+
+
+def raw_to_hex(raw_str):
+    """
+    convert raw bytes to hex string
+    :param raw_str: raw bytes
+    :return: hexadecimal string
+    """
+
+    return binascii.hexlify(raw_str)
+
+
+def str_to_base64(sample_string):
+    """
+    convert any ascii string to base64 encoding
+    :param sample_string: ascii string to be encoded to base 64
+    :return: base64 encoded string
+    """
+
+    sample_string_bytes = sample_string.encode()
+    base64_bytes = base64.b64encode(sample_string_bytes)
+    base64_string = base64_bytes.decode()
+    return base64_string
+
+
+def output_format_enc(bech32_str, out_format):
+    """
+    convert the encoded bech32m string to desired(hex/b64/bin) output format
+    :param bech32_str: valid bech32m encoded string
+    :param out_format: output format for encoded string
+    :return: bech32m string encoded in output format (hex/b64/bin)
+    """
+
+    if out_format == "bech32m":
+        return bech32_str
+
+    elif out_format == "hex":
+        return ascii_to_hex(bech32_str)
+
+    elif out_format == "b64":
+        return str_to_base64(bech32_str)
+
+    elif out_format == "bin":
+        raw_hex = ascii_to_hex(bech32_str)
+        return hex_to_raw(raw_hex)
+
+    else:
+        print("Invalid Output Format. Please Specify hex, b64 or bin")
+        sys.exit(0)
+
+
+def output_format_dec(hex_str, out_format):
+    """
+    convert the decoded string to desired(hex/b64/bin) output format
+    :param hex_str: hex string generated after decoding a bech32m string
+    :param out_format: output format for decoded string
+    :return: decoded string in required encoding (hex/b64/bin)
+    """
+
+    if out_format == "hex":
+        return hex_str
+    elif out_format == "b64":
+        return hex_to_base64(hex_str)
+    elif out_format == "bin":
+        return hex_to_raw(hex_str)
+    else:
+        print("Invalid Output Format. Please Specify hex, b64 or bin")
+
+
+def read_file(input_file_name, input_format):
+    """
+    read encoded/decoded string/bytes from file as per input format
+    :param input_file_name: file name from which input string/bytes are to be read
+    :param input_format: bin/hex/b64
+    :return: string/bytes as per specified input format
+    """
+
+    try:
+
+        if input_format == "bin":
+            mode = "rb"
+        else:
+            mode = "r"
+        with open(input_file_name, mode) as f:
+            input_data = f.read()
```

Comment:
issue (blocking): The `f.read()` with "r" mode crashes the application if a non utf-8 character is in a file.